### DISCUSSION TITLE
Integrating Supabase Realtime with NestJS

### DISCUSSION THREAD

**Discussion by bryan-vh** [supabase/supabase#11386] [Questions] [answered]
Hey all, this is my first time using Supabase.

I wanted to know if it seems feasible to use Supabase Realtime inside of NestJS to listen for changes in a table. The reason I am trying to do this is because I am using NestJS as my WS / REST API for a mobile client. In this case, I wanted to create a WebSocket in NestJS that this mobile client can listen to / interact with, but wanted to be able to propagate the table changes from the `RealtimeClient` to this mobile client.

Currently, I am planning to pass in the `WebSocket` class provided by the `ws` js library to the callback in here:

```
this.realtime
      .channel('public:message')
      .on(
        'postgres_changes',
        { event: 'INSERT', schema: 'public', table: 'message' },
        (payload) => {
          // insert websocket to send payload to here
          console.log(payload);
        },
      )
      .subscribe();
```

I understand that there's an iOS library that's being worked on, but it doesn't seem ready at this time, and I like the flexibility of being able to do other operations in my custom backend. I'd like to know if there's a better way to do this or if what I'm proposing makes sense. Thank you!

**Comment by w3b6x9** [comment, marked as answer]
This was discussed in https://github.com/supabase/realtime/issues/398.
